packages/theming/atlas/scripts/build.js

```diff
@@ -1,65 +1,82 @@
-const { rm, cd } = require("shelljs");
-const { join } = require("path");
 const concurrently = require("concurrently");
+const { access } = require("fs").promises;
+const { join } = require("path");
+const { rm } = require("shelljs");
 
-const mode = process.argv[2] || "build";
-const MX_PROJECT_PATH = process.env.ATLAS_MX_PROJECT_PATH; // should be an absolute path.
-let outputProjectDir = MX_PROJECT_PATH ? MX_PROJECT_PATH : join(__dirname, "../tests/testProject");
-let outputThemeDir;
-let outputThemeSourceDir;
-let projectDeployDirWeb;
+main().catch(e => {
+    console.error(e);
+    process.exit(-1);
+});
 
-switch (mode) {
-    case "build":
-    case "start":
-        outputThemeDir = join(outputProjectDir, "theme");
-        outputThemeSourceDir = join(outputProjectDir, "themesource/atlas_ui_resources");
-        projectDeployDirWeb = join(outputProjectDir, "deployment/web");
-        break;
-    case "release":
-        outputThemeDir = join(__dirname, "../dist/theme");
-        outputThemeSourceDir = join(__dirname, "../dist/themesource");
-        break;
-}
+async function main() {
+    const mode = process.argv[2] || "build";
+
+    let outputDir;
+
+    if (mode === "build" || mode === "start") {
+        const MX_PROJECT_PATH = process.env.ATLAS_MX_PROJECT_PATH; // should be an absolute path.
+        outputDir = MX_PROJECT_PATH ? MX_PROJECT_PATH : join(__dirname, "../tests/testProject");
+
+        const toRemoveDirs = [join(outputDir, "theme"), join(outputDir, "themesource/atlas_ui_resources")];
+        const toRemoveDirsExist = await Promise.all(toRemoveDirs.map(async dir => exists(dir)));
+
+        if (toRemoveDirsExist.includes(true)) {
+            rm("-rf", toRemoveDirs);
+            console.info(`Successfully removed Atlas theme files from Mendix project`);
+        }
+    } else if (mode === "release") {
+        outputDir = join(__dirname, "../dist");
 
-console.info(`Building for ${mode}...`);
-const watchArg = mode === "start" ? "--watch" : "";
-cd(join(__dirname, ".."));
-rm("-rf", outputThemeDir);
-concurrently(
-    [
-        {
-            name: "web-theme-content",
-            command: `copy-and-watch ${watchArg} "src/theme/web/**/*" "${outputThemeDir}/web"`
-        },
-        {
-            name: "web-themesource-content",
-            command: `copy-and-watch ${watchArg} 'src/themesource/atlas_ui_resources/web/**/*' '${outputThemeSourceDir}/web'`
-        },
-        {
-            name: "native-typescript",
-            command: `tsc ${watchArg} --project tsconfig.json --outDir '${outputProjectDir}'`
-        },
-        {
-            name: "native-design-properties-and-manifest",
-            command: `copy-and-watch ${watchArg} 'src/themesource/atlas_ui_resources/native/**/*.json' '${outputThemeSourceDir}/native'`
+        if (await exists(outputDir)) {
```

Comment:
Same here. We can bypass this check and avoid an extra fs operation.

Reply:
See reply on the other `rm` comment.